modules/subscriber/lambda.tf

```diff
@@ -0,0 +1,27 @@
+resource "aws_lambda_event_source_mapping" "subscriber_sqs" {
+  event_source_arn = aws_sqs_queue.queue.arn
+  function_name    = aws_lambda_function.subscriber.arn
+}
+
+resource "aws_lambda_function" "subscriber" {
+  function_name = var.name
+  s3_bucket     = local.parsed_s3_uri["bucket"]
+  s3_key        = local.parsed_s3_uri["key"]
+  role          = aws_iam_role.subscriber.arn
+  handler       = "bootstrap"
+  runtime       = "provided.al2"
+  architectures = ["arm64"]
+  environment {
+    variables = {
+      FILTER_NAME             = local.set_default_filter_name ? "observe-logs-subscription" : var.filter_name
```

Comment:
The `set_default_filter_name` pattern is needed in cloudformation, because we can't tell if a value was set by user or carried over as a default between upgrades. Here you can just set `var.filter_name` directly.